devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/tooling/GradleApplicationModelBuilder.java

```diff
@@ -462,6 +464,38 @@ private static void initProjectModule(Project project, WorkspaceModule.Mutable m
             });
         });
 
+        // This "try/catch" is needed because of the way the "quarkus-cli" Gradle tests work. Without it, the tests fail.
+        try {
+            Class.forName("org.jetbrains.kotlin.gradle.tasks.KotlinJvmCompile");
+            project.getTasks().withType(KotlinJvmCompile.class, t -> {
+                if (!t.getEnabled()) {
+                    return;
+                }
+                final FileTree source = t.getSources().getAsFileTree();
+                if (source.isEmpty()) {
+                    return;
+                }
+                final File destDir = t.getDestinationDirectory().getAsFile().get();
+                if (!allClassesDirs.contains(destDir)) {
+                    return;
+                }
+                final List<File> srcDirs = new ArrayList<>(1);
+                source.visit(a -> {
+                    // we are looking for the root dirs containing sources
+                    if (a.getRelativePath().getSegments().length == 1) {
+                        final File srcDir = a.getFile().getParentFile();
+                        if (srcDirs.add(srcDir)) {
```

Comment:
This is always `true`

Reply:
Yes, IntelliJ gave me this inspection

I noticed that also `srcDirs` is not used, too. I originally was going to include another change, refactoring it in all the places it is repeated to the below:


```java
source.visit(a -> {
    // we are looking for the root dirs containing sources
    if (a.getRelativePath().getSegments().length == 1) {
        final File srcDir = a.getFile().getParentFile();
        DefaultSourceDir sources = new DefaultSourceDir(srcDir.toPath(), destDir.toPath(),
                Collections.singletonMap("compiler", t.getName()));
        sourceDirs.add(sources);
    }
});
```

But I'm not sure why this is the way it is, I didn't want to assume and refactor in case it was doing some magic I didn't understand 🙈

Reply:
It sounds like it was supposed to be a `HashSet` at some point, but I'm +1 in removing it as it doesn't add any value besides increasing the consumed memory when running 😉

Reply:
Got it, I can refactor it no problem then.

I might need help with squashing the commits at the end though.
Last time, G. Smet asked me to do that, and I managed to royally screw up the branch 😅

Reply:
in your local workspace, you can do `git rebase -i main` and then `squash` all but the first commit.  there will be directions on how to do that in the commit message edit phase.  You'll also get a chance to include or remove all the messages of the smashed commits.  I generally don't include them but it depends on what they are.  then you can `git push -f` and your PR will have 1 commit to merge in to main.

Reply:
Bless, ty Evan! 🙏

Reply:
Okey I tried the squash, it seemed to mostly work but there is still the original commit in there for some reason 🤔 
Git makes me feel like I have a room temperature IQ.